vsintegration/src/FSharp.Editor/BraceMatchingService.fs

```diff
@@ -3,117 +3,42 @@
-        if position < 0 || position >= sourceText.Length then
-            None
-        else
-            let shouldBeIgnored(characterPosition) =
-                let textSpan = TextSpan(characterPosition, 1)
-                let classifiedSpans = FSharpColorizationService.GetColorizationData(sourceText, textSpan, fileName, defines, cancellationToken)
-                if classifiedSpans.Any() then
-                    ignoredClassificationTypes |> Seq.contains (classifiedSpans.First().ClassificationType)
-                else
-                    false
-                
-            if shouldBeIgnored(position) then
-                None
-            else
-                let currentCharacter = sourceText.[position]
-
-                let proceedToStartOfString(i) = i - 1
-                let proceedToEndOfString(i) = i + 1
 
-                let afterEndOfString(i) = i >= sourceText.Length
-                let beforeStartOfString(i) = i < 0
-
-                let pickBraceType(leftBrace, rightBrace) =
-                    if currentCharacter = leftBrace then Some(proceedToEndOfString, afterEndOfString, leftBrace, rightBrace)
-                    else if currentCharacter = rightBrace then Some(proceedToStartOfString, beforeStartOfString, rightBrace, leftBrace)
-                    else None
-
-                match supportedBraceTypes |> List.tryPick(pickBraceType) with
-                | None -> None
-                | Some(proceedFunc, stoppingCondition, matchedBrace, nonMatchedBrace) ->
-                    let mutable currentPosition = proceedFunc position
-                    let mutable result = None
-                    let mutable braceDepth = 0
-
-                    while result.IsSome = false && stoppingCondition(currentPosition) = false do
-                        cancellationToken.ThrowIfCancellationRequested()
-                        if shouldBeIgnored(currentPosition) = false then
-                            if sourceText.[currentPosition] = matchedBrace then
-                                braceDepth <- braceDepth + 1
-                            else if sourceText.[currentPosition] = nonMatchedBrace then
-                                if braceDepth = 0 then
-                                    result <- Some(BraceMatchingResult(TextSpan(min position currentPosition, 1), TextSpan(max position currentPosition, 1)))
-                                else
-                                    braceDepth <- braceDepth - 1
-                        currentPosition <- proceedFunc currentPosition
-                    result
+    static member GetBraceMatchingResult(sourceText, fileName, options, position) = async {
+        let isPositionInRange(range) =
+            let span = CommonRoslynHelpers.FSharpRangeToTextSpan(sourceText, range)
+            span.Start <= position && position <= span.End
+        let! matchedBraces = FSharpChecker.Instance.MatchBracesAlternate(fileName, sourceText.ToString(), options)
```

Comment:
> FSharpChecker.Instance [](start = 28, length = 23)

Is FSharpChecker an immutable type?  If not and it contains mutable state consider passing this as a parameter.